apps/web/src/components/providers/EnvironmentProvider.tsx

```diff
@@ -88,6 +88,12 @@ export function EnvironmentProvider({ children }: { children: React.ReactNode })
     selectEnvironment(environments, getEnvironmentId())
   );
 
+  useEffect(() => {
+    if (environments) {
+      setCurrentEnvironment(selectEnvironment(environments, getEnvironmentId()));
+    }
+  }, [environments]);
```

Comment:
this allows to update the `{environment}` content returned by the useEnvironments hook. Without it, it's never resassigned on change

Reply:
Oh, let me check this out! i think it's because of the early termination there if the _id is not changed.